### DISCUSSION TITLE
How to get a list of Playlists from a Channel ID

### DISCUSSION THREAD

**Discussion by BourgeoisDirk** [Tyrrrz/YoutubeExplode#773] [Q&A] [answered]
I'm trying to download by playlist from a channel, but can't seem to find anything that will get me a list of playlists from given Channel ID.
Is there someway i can get the playlists from a given channel ?

**Comment by Tyrrrz** [comment]
No, there is no way

**Reply by BourgeoisDirk** [reply]
Then may i add it as a feature request ?

I take it then that a search on channel id would not work ?
`var playlistSearch = await Core.Client.Search.GetPlaylistsAsync(ChannelId, Core.ProcessCancellationToken.Token);
var playlists = playlistSearch.Cast<YoutubeExplode.Playlists.Playlist>().ToArray();`

**Reply by BourgeoisDirk** [reply]
I guess it could be done backwards for the time being, until there is a Channel.Playlists.GetAsync() ?

public async Task LoadAsync(YoutubeExplode.Channels.Channel channel)
{
    var videos = await Core.Client.Channels.GetUploadsAsync(channel.Id, Core.ProcessCancellationToken.Token);
    var playlistIds = videos.Select(x => x.PlaylistId).Distinct().Select(x => x.Value).ToList();

    var playlists = new Dictionary<string, YoutubeExplode.Playlists.Playlist>();
    foreach (var playlistId in playlistIds)
        playlists.Add(playlistId, await Core.Client.Playlists.GetAsync(playlistId, Core.ProcessCancellationToken.Token));
}

**Reply by Tyrrrz** [reply]
I don't want to extend the scope of the project any more. Your approach will be quite slow for large channels and I don't know of anything similar to the special "channel uploads" playlist for playlists.

https://wiki.archiveteam.org/index.php/YouTube/Technical_details#Playlists

**Reply by BourgeoisDirk** [reply]
Hey, i have a problem with Building a Channel Hierarchy from the Videos from a given Channel.

var videos = await Core.Client.Channels.GetUploadsAsync(channel.Id, Core.ProcessCancellationToken.Token);
var playlistIds = videos.Select(x => x.PlaylistId).Distinct().Select(x => x.Value).ToList();

It appears that for some reason, after getting the channel by Handle, and then getting the list of Videos via Channels.GetUploadsAsync(channelId), that all of the video's have the same PlaylistId ❌ 
Is there any way to build a Hierarchy from a Channel object ? Does not matter if it is slow, i just need to build a list of the PlaylistID's that belong to given Channel.

Kind regards

**Reply by Tyrrrz** [reply]
I don't think it's possible what you're trying to do

**Comment by BourgeoisDirk** [comment, marked as answer]
Well then, down to a lower level it is ^^

**Requirements:**
- API Key via Google Console: https://support.google.com/googleapi/answer/6158862?hl=en
- NuGet Package: Google.Apis
- NuGet Package: Google.Apis.YouTube.v3

**Code to fetch list of PlaylistIDs from a ChannelID:**
using Google.Apis.Services;
using Google.Apis.YouTube.v3;

        private async Task<List<string>> GetPlaylistsAsync(string channelId)
        {
            var youtubeService = new YouTubeService(new BaseClientService.Initializer() { ApiKey = "enter api key here" });

            var playlistListRequest = youtubeService.Playlists.List("snippet");
            playlistListRequest.ChannelId = channelId;

            var playlistListResponse = await playlistListRequest.ExecuteAsync();
            var playlists = playlistListResponse.Items.Select(x => x.Id).ToList();

            return playlists ;
        }

Or, basically just make an HttpRequest and parse the response, you can try this out in a browser:
https://www.googleapis.com/youtube/v3/playlists?key=APIKEY&channelId=CHANNELID&part=snippet

After this you can just get the data from each PlaylistID using YoutubeExplode:
var playlist = await YoutubeExplode.YoutubeClient.Playlists.GetAsync(playlistId, ProcessCancellationToken)
var videos = await YoutubeExplode.YoutubeClient.Playlists.GetVideosAsync(playlist.Id, ProcessCancellationToken)


Keep up the good work !
Kind regards

**Comment by h0lg** [comment]
Great to find this discussion - the same question came up downstream!
I guess this is an interesting feature for channels that host playlists created from videos of *other* channels - those videos wouldn't pop up in the channel's `Uploads` playlist.

If all you need is a **script you can paste into your browser console** to get the playlists from a channel's `/playlists` page, [check this out](https://github.com/h0lg/SubTubular/issues/16#issuecomment-3340495566).
